lib/webpacker/source.rb

```diff
@@ -1,23 +1,30 @@
+require "webpacker/manifest"
+require "webpacker/package_json"
+
 # Translates a logical reference for a pack source into the final
 # path needed in the HTML using generated digests.json manifest.
-
 class Webpacker::Source
+  class SourceError < StandardError; end
+
   def initialize(filename)
     @filename = filename
+    @config ||= Webpacker::PackageJson.new
   end
 
   def path
-    if config[:dev_server_host].present?
-      "#{config[:dev_server_host]}/#{filename}"
+    if Rails.env.development? && config.dev_server[:enabled]
+      "http://#{config.dev_server[:host]}:#{config.dev_server[:port]}/#{filename}"
     else
-      Webpacker::Digests.lookup(filename)
+      Webpacker::Manifest.load(digests_path)
```

Comment:
We should only be reloading this during development mode. RAILS_ENV == 'production' ==> no need to reload.